### PR TITLE
Add Argo Controller Unit tests and Linting

### PR DESCRIPTION
**Pull request by DomFleischmann**

### CODE REVIEW

File: operators/argo-controller/src/charm.py [hunk 1]
```diff
@@ -53,7 +51,7 @@ def main(self, event):
 
         if not self.minio.is_available:
             self.log.info("Waiting for Minio data")
-            self.model.unit.status = MaintenanceStatus("Waiting for Minio data")
+            self.model.unit.status = WaitingStatus("Waiting for Minio data")
```
Comment (knkski) [inline operators/argo-controller/src/charm.py:54]: Could this be updated to `Waiting for MinIO connection information` or similar? `data` is a little bit unclear.

File: operators/argo-controller/test/unit/test_charm.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            "registrypath": "argoproj/workflow-controller:v2.3.0",
+            "username": "",
+            "password": "",
+        },
+    )
+    harness.begin_with_initial_hooks()
+    pod_spec = harness.get_pod_spec()
+
+    # confirm that we can serialize the pod spec
+    yaml.dump(pod_spec, Dumper=_DefaultDumper)
```
Comment (knkski) [inline operators/argo-controller/test/unit/test_charm.py:46]: Is there a reason to not just use `yaml.safe_dump`?

File: .github/workflows/check-formatting.yaml [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
+name: Lint Code and Unit Tests 
 
 on:
   - push
   - pull_request
 
 jobs:
   build:
-    name: Lint Code
+    name: Lint Code and Unit Tests
```
Comment (knkski) [inline .github/workflows/check-formatting.yaml:9]: Would it make sense to have this be a separate job? e.g.

```yaml
jobs:
  lint: ...
  test: ...
```

File: operators/argo-controller/tox.ini [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[flake8]
```
Comment (knkski) [inline operators/argo-controller/tox.ini:1]: Is this file also necessary in addition to the root-level tox.ini?

File: operators/argo-controller/test/unit/test_charm.py (hunk 2 above)
Comment (DomFleischmann) [reply operators/argo-controller/test/unit/test_charm.py:46]: You are right that would probably be better, I'll look into that and probably also change it in the interface

File: operators/argo-controller/tox.ini (hunk 4 above)
Comment (DomFleischmann) [reply operators/argo-controller/tox.ini:1]: I can move it all to the root-level tox.ini. I just thought it would be cleaner this way.

Comment (knkski) [reply operators/argo-controller/tox.ini:1]: Ah, sure. I'd probably move it to the root level `tox.ini` myself, but that's also not a strong preference.

File: operators/argo-controller/test/unit/test_charm.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import pytest
+
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+from ops.testing import Harness
+import yaml
+
+from charm import ArgoControllerCharm
+
+
+if yaml.__with_libyaml__:
```
Comment (knkski) [inline operators/argo-controller/test/unit/test_charm.py:10]: I think this is no longer necessary with the use of `yaml.safe_dump` below?

**Review by knkski — APPROVED** [review]
Minor nit, but otherwise LGTM

**Pull request merged**